src/app-layer.c

```diff
@@ -159,6 +190,22 @@ void AppLayerIncInternalErrorCounter(ThreadVars *tv, Flow *f)
     }
 }
 
+static void AppLayerIncrErrorExcPolicyCounter(ThreadVars *tv, Flow *f, enum ExceptionPolicy policy)
+{
+    uint16_t id = 0;
+    /* for the summary values */
+    uint16_t g_id = 0;
+    id = applayer_counters[f->protomap][f->alproto].eps_error.eps_id[policy];
+    g_id = eps_error_summary.eps_id[policy];
+
+    if (likely(tv && id > 0)) {
```

Comment:
nit: checking for tv twice, and also doing work before it. Seems we need this check at the start of the func.

Also, can it only happen in unittests? If so, we can ifdef wrap it.

Reply:
> nit: checking for tv twice, and also doing work before it. Seems we need this check at the start of the func.
> 
roger.
> Also, can it only happen in unittests? If so, we can ifdef wrap it.
I thought this was a way to check whether the counters existed or not - which might happen if the exception policy counters weren't registered, right? I'll try wrapping up and check what happens...